tests/AgentTests/FunctionalAPITests.hs

```diff
@@ -115,6 +115,11 @@ functionalAPITests t = do
       testAsyncCommandsRestore t
     it "should accept connection using async command" $
       withSmpServer t testAcceptContactAsync
+  describe "Queue rotation" $ do
+    it "should switch delivery to the new queue with one server" $
+      testSwitchConnection t initAgentServers
+    it "should switch delivery to the new queue with 2 servers" $
```

Comment:
we should probably add tests that rotation completes with client restarts and/or server restarts